graphql_compiler/query_formatting/common.py

```diff
@@ -377,4 +183,23 @@ def insert_arguments_into_query(compilation_result: CompilationResult, arguments
         )
 
 
+def deserialize_argument(name: str, expected_type: QueryArgumentGraphQLType, value: Any,) -> Any:
+    """Deserialize a GraphQL argument, raising a GraphQLInvalidArgumentError if invalid."""
+    try:
+        return deserialize_value(expected_type, value)
+    except (ValueError, TypeError, AssertionError) as e:
```

Comment:
I would not catch an `AssertionError` here. We want to raise `GraphQLInvalidArgumentError` if there is an "user error", An `AssertionError` is more indicative of a server-side/compiler bug.